numpy/core/tests/test_scalar_ctors.py

```diff
@@ -37,6 +39,39 @@ def test_floating_overflow(self):
         assert_equal(flongdouble, -np.inf)
 
 
+class TestExtraArgs:
+    def test_superclass(self):
+        # try both positional and keyword arguments
+        s = np.str_(b'\\x61', encoding='unicode-escape')
+        assert s == 'a'
+        s = np.str_(b'\\x61', 'unicode-escape')
+        assert s == 'a'
+
+        # previously this would return '\\xx'
+        with pytest.raises(UnicodeDecodeError):
+            np.str_(b'\\xx', encoding='unicode-escape')
+        with pytest.raises(UnicodeDecodeError):
+            np.str_(b'\\xx', 'unicode-escape')
+
+        # superclass fails, but numpy succeeds
+        assert np.bytes_(-2) == b'-2'
+
+    def test_datetime(self):
+        dt = np.datetime64('2000-01', ('M', 2))
+        assert np.datetime_data(dt) == ('M', 2)
```

Comment:
Fatal Python error: Segmentation fault
Current thread 0x00007fb7c6f6e740 (most recent call first):
  File "/home/travis/build/numpy/numpy/numpy/core/tests/test_scalar_ctors.py", line 61 in test_datetime
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/python.py", line 167 in pytest_pyfunc_call
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/python.py", line 1445 in runtest
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 132 in pytest_runtest_call
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 208 in <lambda>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 235 in from_call
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 208 in call_runtest_hook
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 183 in call_and_report
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 97 in runtestprotocol
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/runner.py", line 82 in pytest_runtest_protocol
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/main.py", line 270 in pytest_runtestloop
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/main.py", line 246 in _main
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/main.py", line 196 in wrap_session
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/main.py", line 239 in pytest_cmdline_main
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 93 in main
  File "/home/travis/build/numpy/numpy/numpy/_pytesttester.py", line 188 in __call__
```
This doesn't look related to my change, but rather to the fact that `datetime_data` is seemingly broken

Reply:
I think `array_datetime_data` must DECREF `dtype` later, so that would explain why a seemingly unrelated change can make it break reliably probably.

Reply:
OK, I tried and it fixes things. Will just push a commit here, I hope you do not mind.

Reply:
Nice find! I suspect my mistake was using an exotic dtype that isn't really used much.